build.xml

```diff
@@ -156,7 +156,10 @@
     <property name="jamm.version" value="0.3.2"/>
     <property name="ecj.version" value="3.33.0"/>
     <property name="ohc.version" value="0.5.1"/>
-    <property name="asm.version" value="7.1"/>
+    <!-- When updating ASM, please, do consider whether you might need to update also FBUtilities#ASM_BYTECODE_VERSION
+       and the simulator InterceptClasses#BYTECODE_VERSION, in particular if we are looking to provide Cassandra support
+       for newer JDKs (CASSANDRA-17873). -->
+    <property name="asm.version" value="9.4"/>
```

Comment:
I copied the comment as-is, but the simulator reference is irrelevant, we might want to remove it.

Reply:
Yes, the simulator comment is irrelevant, as the simulator has not been added to this version of Cassandra.
Also, in Cassandra, we update for newer JDK builds, too, so we bump already. Though I agree with your other comment about forgetting to override methods, etc., it is valid. Can we add a note that we need to override that one? Also, why not override already? What do I miss? I suspect that in 5.0/main-5.0, that may have been just missed or left for later, as you said, and forgotten. Anyway, we still do not need it as we are still building with 11.

Reply:
Also, I would recommend version 9.7 or later. I will add more info to the ticket.

Reply:
> Though I agree with your other comment about forgetting to override methods, etc., it is valid. Can we add a note that we need to override that one? Also, why not override already?

Do you mean override the methods?

Reply:
Yes

Reply:
If we're already on `Opcodes.ASM9`, overriding the methods won't make a difference because the parent class implementations of those methods are no-ops.

Maybe I'm overthinking this, it should be covered by tests anyway. In the context of UDFs, ASM is used for sandboxing, I see this is covered in `UFSecurityTest.testSecurityPermissions()`. So whenever UDFs support a new JDK version, new cases should be added there to check that UDFs that use forbidden operations in "new" places are properly being rejected.